bin/collator/src/parachain/shell_upgrade.rs

```diff
@@ -21,12 +21,21 @@
 use astar_primitives::*;
 use cumulus_client_consensus_common::{ParachainCandidate, ParachainConsensus};
 use cumulus_primitives_core::relay_chain::{Hash as PHash, PersistedValidationData};
+use cumulus_primitives_core::BlockT;
+use cumulus_test_relay_sproof_builder::RelayStateSproofBuilder;
+use fc_rpc::pending::ConsensusDataProvider;
 use futures::lock::Mutex;
+use sc_client_api::{AuxStore, UsageProvider};
 use sc_consensus::{import_queue::Verifier as VerifierT, BlockImportParams, ForkChoiceStrategy};
-use sp_api::ApiExt;
+use sp_api::{ApiExt, ProvideRuntimeApi};
+use sp_consensus_aura::digests::CompatibleDigestItem;
+use sp_consensus_aura::sr25519::AuthoritySignature;
 use sp_consensus_aura::{sr25519::AuthorityId as AuraId, AuraApi};
```

Comment:
Nitpick comment - you specify `sp_consensus_aura` 3 times, with `sr25519` being specified twice. Same for few other crates. IMO, would be better to group them.